src/helper/ecdsa.rs

```diff
@@ -0,0 +1,202 @@
+        EcdsaPublicParams, Fingerprint, KeyDetails, KeyId, KeyVersion, Mpi, Password,
+        PublicKeyTrait, PublicParams, SecretKeyTrait, SignatureBytes,
+    },
+};
+
+impl<C> PgpPublicKey for ecdsa::VerifyingKey<C>
+where
+    Self: PgpEcdsaPublicKey,
+    C: PrimeCurve + CurveArithmetic,
+{
+    const PGP_ALGORITHM: PublicKeyAlgorithm = PublicKeyAlgorithm::ECDSA;
+
+    fn pgp_parameters(&self) -> PublicParams {
+        let key = self.ecdsa_public_key();
+        PublicParams::ECDSA(key)
+    }
+}
+
+/// Encoding for an ecdsa public key
+pub trait PgpEcdsaPublicKey {
+    /// public key parameter for a given public key
+    fn ecdsa_public_key(&self) -> EcdsaPublicParams;
+}
+
+impl PgpEcdsaPublicKey for p256::ecdsa::VerifyingKey {
+    fn ecdsa_public_key(&self) -> EcdsaPublicParams {
+        let key = self.into();
+        EcdsaPublicParams::P256 { key }
+    }
+}
+
+/// [`signature::Signer`] backed signer for PGP.
+pub struct EcdsaSigner<T, C> {
+    inner: T,
+    public_key: PublicKey,
+    _signature: PhantomData<C>,
+}
+
+impl<C, T> EcdsaSigner<T, C>
+where
+    T: Keypair,
+    T::VerifyingKey: PgpPublicKey,
+{
+    /// Create a new signer with a given public key
+    pub fn new(inner: T, created_at: DateTime<Utc>) -> Result<Self> {
+        let public_key = PubKeyInner::new(
+            KeyVersion::V4,
+            <T as Keypair>::VerifyingKey::PGP_ALGORITHM,
+            created_at,
+            None,
+            inner.verifying_key().pgp_parameters(),
+        )?;
+        let public_key = PublicKey::from_inner(public_key)?;
+
+        Ok(Self {
+            inner,
+            public_key,
+            _signature: PhantomData,
+        })
+    }
+}
+
+impl<C, T> Keypair for EcdsaSigner<T, C>
+where
+    T: Keypair,
+{
+    type VerifyingKey = T::VerifyingKey;
+
+    fn verifying_key(&self) -> Self::VerifyingKey {
+        self.inner.verifying_key()
+    }
+}
+
+impl<C, T> EcdsaSigner<T, C>
+where
+    C: PrimeCurve + DigestPrimitive,
+    SignatureSize<C>: ArrayLength<u8>,
+    T: PrehashSigner<ecdsa::Signature<C>>,
+    C::Digest: KnownDigest,
+{
+    fn sign_prehash(&self, hash: HashAlgorithm, prehash: &[u8]) -> Result<Vec<Vec<u8>>> {
+        if C::Digest::HASH_ALGORITHM != hash {
+            bail!(
+                "Signer only support {expected:?}, found {found:?}",
+                expected = C::Digest::HASH_ALGORITHM,
+                found = hash
+            );
+        }
+
+        if <C::Digest as OutputSizeUser>::OutputSize::USIZE != prehash.len() {
+            bail!(
+                "Signer expected a hash of length ({expected} bytes), found ({found} bytes)",
+                expected = <C::Digest as OutputSizeUser>::OutputSize::USIZE,
+                found = prehash.len()
+            );
+        }
+
+        let signature = self.inner.sign_prehash(prehash)?;
+        let (r, s) = signature.split_bytes();
+        Ok(vec![r.to_vec(), s.to_vec()])
+    }
+}
+
+impl<C, T> fmt::Debug for EcdsaSigner<T, C> {
```

Comment:
we now have `derive_more` in the deps, so please use that for custom debug impls